apps/stripe/src/app/api/saleor/transaction-refund-requested/use-case.ts

```diff
@@ -0,0 +1,176 @@
+import { err, ok, Result } from "neverthrow";
+
+import { TransactionRefundRequestedEventFragment } from "@/generated/graphql";
+import { createLogger } from "@/lib/logger";
+import { AppConfigRepo } from "@/modules/app-config/repositories/app-config-repo";
+import { SaleorApiUrl } from "@/modules/saleor/saleor-api-url";
+import { SaleorMoney } from "@/modules/saleor/saleor-money";
+import {
+  AppIsNotConfiguredResponse,
+  BrokenAppResponse,
+  MalformedRequestResponse,
+} from "@/modules/saleor/saleor-webhook-responses";
+import { mapStripeErrorToApiError } from "@/modules/stripe/stripe-api-error";
+import { StripeMoney } from "@/modules/stripe/stripe-money";
+import { createStripePaymentIntentId } from "@/modules/stripe/stripe-payment-intent-id";
+import { IStripeRefundsApiFactory } from "@/modules/stripe/types";
+import { RefundFailureResult } from "@/modules/transaction-result/refund-result";
+
+import {
+  TransactionRefundRequestedUseCaseResponses,
+  TransactionRefundRequestedUseCaseResponsesType,
+} from "./use-case-response";
+
+type UseCaseExecuteResult = Result<
+  TransactionRefundRequestedUseCaseResponsesType,
+  AppIsNotConfiguredResponse | BrokenAppResponse | MalformedRequestResponse
+>;
+
+export class TransactionRefundRequestedUseCase {
+  private logger = createLogger("TransactionRefundRequestedUseCase");
+  private appConfigRepo: AppConfigRepo;
+  private stripeRefundsApiFactory: IStripeRefundsApiFactory;
+
+  constructor(deps: {
+    appConfigRepo: AppConfigRepo;
+    stripeRefundsApiFactory: IStripeRefundsApiFactory;
+  }) {
+    this.appConfigRepo = deps.appConfigRepo;
+    this.stripeRefundsApiFactory = deps.stripeRefundsApiFactory;
+  }
+
+  async execute(args: {
+    appId: string;
+    saleorApiUrl: SaleorApiUrl;
+    event: TransactionRefundRequestedEventFragment;
+  }): Promise<UseCaseExecuteResult> {
+    const { appId, saleorApiUrl, event } = args;
+
+    // Additional validation as Saleor Graphql schema doesn't require these fields and they are needed to process the event
+    if (!event.transaction) {
+      this.logger.error("Transaction not found in event", {
+        event,
+      });
+
+      return err(new MalformedRequestResponse());
+    }
```

Comment:
Since this is work around wrong schema, we can refactor this (in the other pr) to be actually 
1. Throw
2. Sentry (just in case)
3. Assert even HAS transaction

and we can even do that in route, or in use cases, so we reduce some boilerplate
we can still throw MaformedRequestResponse, it just doesnt have to be neverthrow